ARCHITECTURE.md

```diff
@@ -41,49 +31,201 @@ Pay attention to the "**Architecture Invariant**" sections.
-- `crates/ironrdp-testsuite-core`: contains all integration tests for code living in the core tier, in a single binary,
-  organized in modules. **Architectural Invariant**: no dependency from another tier is allowed.
-  It must be the case that compiling and running the core test suite does not require building any library from
-  the extra tier. This is to keep iteration time short.
-- `crates/ironrdp-testsuite-extra`: contains all integration tests for code living in the extra tier, in a single binary,
-  organized in modules. (WIP: this crate does not exist yet.)
-- `crates/ironrdp-fuzzing`: provides test case generators and oracles for use with fuzzing.
-- `fuzz`: fuzz targets for code in core tier.
-- `xtask`: IronRDP’s free-form automation using Rust code.
-
 **Architecture Invariant**: these crates are not, and will never be, an **API Boundary**.
 
+#### [`crates/ironrdp-pdu-generators`](./crates/ironrdp-pdu-generators)
+
+`proptest` generators for `ironrdp-pdu` types.
+
+#### [`crates/ironrdp-session-generators`](./crates/ironrdp-session-generators)
+  
+`proptest` generators for `ironrdp-session` types.
+
+#### [`crates/ironrdp-testsuite-core`](./crates/ironrdp-testsuite-core)
+
+Contains all integration tests for code living in the core tier, in a single binary, organized in modules.
+  
+**Architectural Invariant**: no dependency from another tier is allowed. It must be the case that
+compiling and running the core test suite does not require building any library from the extra tier.
+This is to keep iteration time short.
+
+#### [`crates/ironrdp-testsuite-extra`](./crates/ironrdp-testsuite-extra)
+
+Contains all integration tests for code living in the extra tier, in a single binary, organized in modules.
+
+(WIP: this crate does not exist yet.)
+
+#### [`crates/ironrdp-fuzzing`](./crates/ironrdp-fuzzing)
+
+Provides test case generators and oracles for use with fuzzing.
+
+#### [`fuzz`](./fuzz)
+
+Fuzz targets for code in core tier.
+
+#### [`xtask`](./xtask)
+
+IronRDP’s free-form automation using Rust code.
+
 ### Community Tier
 
-Crates provided and maintained by the community.
-Core maintainers will not invest a lot of time into these.
-One or several community maintainers are associated to each one
+Crates provided and maintained by the community. Core maintainers will not invest a lot of time into
+these. One or several community maintainers are associated to each one.
+
+The IronRDP team is happy to accept new crates but may not necessarily commit to keeping them
+working when changing foundational libraries. We promise to notify you if such a crate breaks, and
+will always try to fix things when it's a minor change.
+
+#### [`crates/ironrdp-acceptor`](./crates/ironrdp-acceptor) (@mihneabuz)
+
+State machines to drive an RDP connection acceptance sequence
+
+#### [`crates/ironrdp-server`](./crates/ironrdp-server) (@mihneabuz)
+
+Extendable skeleton for implementing custom RDP servers.
+
```

Comment:
Hi! @mihneabuz
Since you are the one who contributed them, I figured that maybe you could be the maintainer for these.
Is it okay for you?

Note that I’m willing to move these crates to the extra tier at some point in the future, when we are done consolidating what we already have in the core and extra tiers.

Reply:
Sure, that sounds great! :)

Reply:
Great to hear, thank you!